tests/integration/targets/dcnm_policy/tests/dcnm/dcnm_policy_merge.yaml

```diff
@@ -408,6 +408,74 @@
       loop_control:
         index_var: my_idx
 
+    - name: Delete all created policies
+      cisco.dcnm.dcnm_policy:
+        fabric: "{{ ansible_it_fabric }}"
+        state: deleted                     # only choose form [merged, deleted, query]
+        config:
+          - name: template_101  # This can either be a policy name like POLICY-xxxxx or template name
+          - name: template_102  # This can either be a policy name like POLICY-xxxxx or template name
+          - name: template_103  # This can either be a policy name like POLICY-xxxxx or template name
+          - name: template_104  # This can either be a policy name like POLICY-xxxxx or template name
+          - name: template_105  # This can either be a policy name like POLICY-xxxxx or template name
+          - switch:
+             - ip: "{{ ansible_switch1 }}"
+      register: result
+
+    - assert:
+        that:
+          - 'item["RETURN_CODE"] == 200'
+          - 'item["MESSAGE"] == "OK"'
+      loop: '{{ result.response }}'
+
+
+    - name: Create policies - checking for policies under switch block
+      cisco.dcnm.dcnm_policy:
+        fabric: "{{ ansible_it_fabric }}"
+        deploy: false
+        config:
+          - name: template_101  # This must be a valid template name
+            description: create template_101
+            priority: 101
+
+          - name: template_102  # This must be a valid template name
+            description: create template_102
+            priority: 102
+
+          - name: template_103  # This must be a valid template name
+            description: create template_103
+            priority: 103
+
+          - switch:
+              - ip: "{{ ansible_switch1 }}"
+                policies:
+                  - name: template_101  # This must be a valid template name
+                    description: create template_101
+                    priority: 101
+
+                  - name: template_102  # This must be a valid template name
+                    description: create template_102 - with changes
+                    priority: 202
+
+              - ip: "{{ ansible_switch2 }}"
+                policies:
+                  - name: template_104  # This must be a valid template name
+                    description: create template_104 - with changes
+                    priority: 104
+
+                  - name: template_105  # This must be a valid template name
+                    description: create template_105 - with changes
+                    priority: 105
+      register: result
+
+    - assert:
```

Comment:
For this test case is the following accurate:

* Policy using `template_101` is created on `ansible_switch1`
* Policy using `template_102` is created on `ansible_switch1`
* Policy using `template_103` is cretaed on `ansible_switch1` and `ansible_switch2`
* Policy using `teamplte_104` is created on `ansible_swtich2`
* Policy using `teamplte_105` is created on `ansible_swtich2`

If this is the case can we check to make sure that the policies were deployed on the correct switch ip's in the assert block?

Reply:
Sure let me add the checks. The only thing we need to be aware of is that the response only has serial numbers, and so we need to add these variables to group_vars for us to check. Otherwise this test case may fail. In case we plan to setup dynamic topologies and add random switches, we need to ensure that proper serial numbers are also defined for these in the variables.

Reply:
Regarding the scope of policies, I have interpreted in one way. May be we should discuss this in our meeting on 7th Mar, to finalise the interpretation. The complexity arises in case all policies are same i.e. under config as well as the switch block. We will discuss this and finalise.